solutions/advanced/ac-GameOnTree.mdx

```diff
@@ -0,0 +1,70 @@
+---
+id: ac-GameOnTree
+source: AtCoder Grand Contest
+title: Game On Tree
+author: Aakash Gokhale
+---
+
+[Official Editorial (Japanese)](https://atcoder.jp/contests/agc017/editorial)
+
+Since the game in the problem seems similar to Nim, let's try to use Sprague-Grundy numbers.
+
+To solve the problem, for a certain root with subtrees, $s_1, s_2, ..., s_k$, we can try to convert all these subtrees into a stack of some size.
+Then, by xoring all these stack sizes (like we do in Nim), we can tell whether we win or lose on this tree.
+
+To compute the size of the stack for a subtree, we can recursively solve the problem.
+For a subtree with $1$ node, the size of the stack will be $0$ because there are no edges that can be deleted.
+
+If the grundy value of the root is positive, then Alice wins, otherwise if the grundy value of the root is $0$, Bob wins.
+
+Notice that when we get the grundy numbers for each subtree we need to add $1$ to all of them before xoring.
+This is because each stack's size increases by $1$ when you consider the edge from the root of $s_i$ to the root of the entire tree.
+
+For example consider the tree:
+
+$3 \\ $
+$1$ $2 \\ $
+$1$ $3 \\ $
```

Comment:
come on just make a diagram at this point

Reply:
bump

Reply:
😦

how's this?
![image](https://github.com/user-attachments/assets/a87fee06-b764-436f-ba13-36a9c2225977)

Reply:
fire